### PR TITLE
#1855 : When we fail to parse a "CALL" command provide a better error message

### LINKED ISSUE #1855: Parse errors for "CALL" always state CALL doesn't exist
Since we delegate back to the Spark parser on parsing errors, any errors on our extended methods will error out stating the command doesn't exist.

For example

```
== SQL ==
CALL m spark_catalog.system.migrate('default.table', ('foo' = 'bar'))
^^^

org.apache.spark.sql.catalyst.parser.ParseException: 
mismatched input 'CALL' expecting {'(', 'ADD', 'ALTER', 'ANALYZE', 'CACHE', 'CLEAR', 'COMMENT', 'COMMIT', 'CREATE', 'DELETE', 'DESC', 'DESCRIBE', 'DFS', 'DROP', 'EXPLAIN', 'EXPORT', 'FROM', 'GRANT', 'IMPORT', 'INSERT', 'LIST', 'LOAD', 'LOCK', 'MAP', 'MERGE', 'MSCK', 'REDUCE', 'REFRESH', 'REPLACE', 'RESET', 'REVOKE', 'ROLLBACK', 'SELECT', 'SET', 'SHOW', 'START', 'TABLE', 'TRUNCATE', 'UNCACHE', 'UNLOCK', 'UPDATE', 'USE', 'VALUES', 'WITH'}(line 1, pos 0)

== SQL ==
CALL m spark_catalog.system.migrate('default.table', ('foo' = 'bar'))
^^^
```

When we should instead be reporting the parse specific error if we correctly parse the beginning.

### PR DESCRIPTION
**Pull request by RussellSpitzer**
Previously we would always delegate to the Spark parser, even when a
CALL request could not be parsed. This leads to confusing errors which
would state that "CALL" was not a good first token. This change has us
only using the custom parser for "CALL" commands and delegating for
everything else.

Fixes #1855

### CODE REVIEW

**Comment by RussellSpitzer** [comment]
@aokolnychyi + @rdblue , We could do something more complicated but I think this is probably sufficient since we really don't want to add more parsing commands in the future.

File: spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSparkSqlExtensionsParser.scala [hunk 1]
```diff
@@ -163,9 +163,9 @@ case object IcebergSqlExtensionsPostProcessor extends IcebergSqlExtensionsBaseLi
   }
 
   private def replaceTokenByIdentifier(
-      ctx: ParserRuleContext,
-      stripMargins: Int)(
-      f: CommonToken => CommonToken = identity): Unit = {
+                                        ctx: ParserRuleContext,
+                                        stripMargins: Int)(
+                                        f: CommonToken => CommonToken = identity): Unit = {
```
Comment (rdblue) [inline spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSparkSqlExtensionsParser.scala:168]: Can you revert this whitespace-only change?

**Comment by rdblue** [comment]
Looks great to me, once the whitespace change is fixed!

File: spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSparkSqlExtensionsParser.scala (hunk 1 above)
Comment (RussellSpitzer) [reply spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSparkSqlExtensionsParser.scala:168]: Woops, not sure when that snuck in

Comment (RussellSpitzer) [reply spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSparkSqlExtensionsParser.scala]: Fixed!

**Review by rdblue — APPROVED** [review]
Looks good to me. I'll merge when tests are passing.

**Pull request merged**